api-report/telemetry-utils.api.md

```diff
@@ -323,7 +323,10 @@ export enum TelemetryDataTag {
 }
 
 // @public
-export type TelemetryEventPropertyTypeExt = string | number | boolean | undefined | (string | number | boolean)[];
+export type TelemetryEventPropertyTypeExt = string | number | boolean | undefined | (string | number | boolean)[] | {
```

Comment:
export type primitive = string | number | boolean;
export type primitiveOrArray = primitive | primitive[];
export type TelemetryEventPropertyTypeExt = primitiveOrArray | Record<string, primitiveOrArray>;

?

Reply:
Yeah I started with something like that but didn't really want to change the Public exports of the package. Your types are much more generic and probably ok, more than what I had done.  And yeah I forgot about Record :P 

Feel free to put a PR out to add these types, I'm not thinking I'll take the time unless I'm making changes here again.